src/tpm/macos/provider.rs

```diff
@@ -0,0 +1,193 @@
+    #[instrument]
+    fn load_key(
+        &mut self,
+        _key_id: &str,
+        _config: Box<dyn Any>,
+    ) -> Result<(), SecurityModuleError> {
+        let config = *_config.downcast::<SecureEnclaveConfig>().map_err(|_| SecurityModuleError::InitializationError(("Failed to initialize config").to_owned()))?; 
+        let _ = self.set_config(config.clone());
+        let algorithm = convert_algorithms(config.clone()); 
+        let hash = convert_hash(config.hash.expect("No Hash given"));
+
+        let load_key = apple_secure_enclave_bindings::provider::rust_crypto_call_load_key(_key_id.to_string(), algorithm, hash);
+
+        if Regex::new("(?i)error").unwrap().is_match(load_key.as_str()){
+            Err(SecurityModuleError::InitializationError(load_key.to_string()))
+        } else {
+            Ok(())
+        }
+    }
+
+
+    /// Initializes the Secure Enclave module and returns a handle for cryptographic operations.
+    ///
+    /// This method initializes the Secure Enclave context and prepares it for use. It should be called
+    /// before performing any other operations with the Secure Enclave.
+    /// 
+    /// Uses the rust_crypto_call_sign_data function from the Swift Secure Enclave bindings.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` that, on success, contains `Ok(())`, indicating that the module was initialized successfully.
+    /// On failure, it returns a `SecurityModuleError`.
+    #[instrument]
+    fn initialize_module(&mut self) -> Result<(), SecurityModuleError> {
+        let initialization_result =
+            apple_secure_enclave_bindings::provider::rust_crypto_call_initialize_module();
+
+        match initialization_result {
+            true => Ok(()),
+            false => Err(SecurityModuleError::InitializationError(
+                "Failed to initialize module".to_string(),
+            )),
+        }
+    }
+}
+
+/// Converts the algorithm type to a String.
+/// 
+/// # Arguments
+/// 
+/// * `config` - A `SecureEnclaveConfig` object containing the configuration for the key.
+/// 
+/// # Returns
+/// 
+/// A `String` containing the AsymmetricEncryption algorithm.
+pub fn convert_algorithms(config: SecureEnclaveConfig) -> String {
+    let asym_algorithm_type = match config.asym_algorithm.expect("Invalid config") {
+        // Is only Asymmetric-Algorithm which is working at that time
+        AsymmetricEncryption::Rsa(_) => "RSA".to_string(),
+        AsymmetricEncryption::Ecc(EccSchemeAlgorithm::EcDsa(_)) => "ECDSA".to_string(), 
+        _ => unimplemented!("Only RSA and ECDSA supported") ,
+    };
+
+    asym_algorithm_type
+}
+
+
+/// Converts the Hash algorithm to a String.
+/// 
+/// # Arguments
+/// 
+/// * `hash` - A `Hash` object containing the hash algorithm.
+/// 
+/// # Returns
+/// 
+/// A `String` containing the Hash algorithm.
+pub fn convert_hash(hash: Hash) -> String {
+    match hash {
+        Hash::Sha1 => "SHA1".to_string(),
```

Comment:
See https://www.nist.gov/news-events/news/2022/12/nist-retires-sha-1-cryptographic-algorithm.

SHA1 is considered broken. If there is not a very specific reason to use SHA1, please remove it.